test/functional/p2p_blocksonly.py

```diff
@@ -39,24 +42,22 @@ def run_test(self):
                 'scriptPubKey': prevtx['vout'][0]['scriptPubKey']['hex'],
             }],
         )['hex']
-        assert_equal(self.nodes[0].getnetworkinfo()['localrelay'], False)
-        with self.nodes[0].assert_debug_log(['transaction sent in violation of protocol peer=0']):
-            self.nodes[0].p2p.send_message(msg_tx(FromHex(CTransaction(), sigtx)))
-            self.nodes[0].p2p.wait_for_disconnect()
-            assert_equal(self.nodes[0].getmempoolinfo()['size'], 0)
-
-        # Remove the disconnected peer and add a new one.
-        del self.nodes[0].p2ps[0]
-        self.nodes[0].add_p2p_connection(P2PInterface())
-
-        self.log.info('Check that txs from rpc are not rejected and relayed to other peers')
-        assert_equal(self.nodes[0].getpeerinfo()[0]['relaytxes'], True)
-        txid = self.nodes[0].testmempoolaccept([sigtx])[0]['txid']
-        with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=1'.format(txid)]):
-            self.nodes[0].sendrawtransaction(sigtx)
-            self.nodes[0].p2p.wait_for_tx(txid)
```

Comment:
FWIW: after this removal `wait_for_tx` is no longer used.

Reply:
I've made some change to this PR, so `wait_for_tx` is still used.